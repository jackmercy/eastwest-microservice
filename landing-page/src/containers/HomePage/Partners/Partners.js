import { memo, useMemo } from 'react';
import { useIntl } from 'react-intl';

import { Content } from '../../components';
import { Carousel } from '../../../components';
import BREAKPOINTS from '../../../constants/BREAKPOINTS';

import { partners } from './partners.configs';
import useStyles from './partners.styles';

const Partners = () => {
  const intl = useIntl();
  const classes = useStyles();
  const settings = useMemo(
    () => ({
      slidesToShow: 5,
      slidesToScroll: 5,
      responsive: [
        {
          breakpoint: BREAKPOINTS.maxLg,
          settings: { slidesToShow: 4, slidesToScroll: 4 },
        },
        {
          breakpoint: BREAKPOINTS.maxMd,
          settings: { slidesToShow: 3, slidesToScroll: 3 },
        },
        {
          breakpoint: BREAKPOINTS.maxSm,
          settings: { slidesToShow: 1, slidesToScroll: 1 },
        },
      ],
    }),
    [],
  );

  return (
    <Content className={classes.root}>
      <h2 className={classes.title} data-test="partners-title">
        {intl.formatMessage({ id: 'HOMEPAGE.PARTNERS.TITLE' })}
      </h2>
      <Carousel
        {...settings}
        className={classes.slider}
        dataTest="partners-carousel"
      >
        {partners.map((item) => (
          <div key={item.name}>
            <img
              loading="lazy"
              alt={item.name}
              title={item.name}
              src={item.logo}
              className={classes.logo}
            />
          </div>
        ))}
      </Carousel>
    </Content>
  );
};

export default memo(Partners);
