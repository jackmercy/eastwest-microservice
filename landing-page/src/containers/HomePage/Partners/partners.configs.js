export const partners = [
  { name: 'Corekees', logo: '/static/images/homepage/partner-01.png' },
  { name: 'RNW', logo: '/static/images/homepage/partner-02.png' },
  { name: 'KoiQuestion', logo: '/static/images/homepage/partner-03.png' },
  { name: 'Phoenix', logo: '/static/images/homepage/partner-04.png' },
  { name: 'Voicepick', logo: '/static/images/homepage/partner-05.png' },
  { name: 'TymeBank', logo: '/static/images/homepage/partner-06.png' },
  { name: 'Quadriz', logo: '/static/images/homepage/partner-07.png' },
  { name: 'Toiya', logo: '/static/images/homepage/partner-08.png' },
  { name: 'Dias Advisors', logo: '/static/images/homepage/partner-09.png' },
  { name: 'Duality', logo: '/static/images/homepage/partner-10.png' },
  { name: '5QLive', logo: '/static/images/homepage/partner-11.png' },
  { name: 'Media 21', logo: '/static/images/homepage/partner-12.png' },
  { name: 'GambleCoin', logo: '/static/images/homepage/partner-13.png' },
  { name: 'Microad', logo: '/static/images/homepage/partner-14.png' },
  { name: 'Grad Greenhouse', logo: '/static/images/homepage/partner-15.png' },
  { name: 'Pycogroup', logo: '/static/images/homepage/partner-16.png' },
  { name: 'Seleno', logo: '/static/images/homepage/partner-17.png' },
  { name: 'Dagiac', logo: '/static/images/homepage/partner-18.png' },
  { name: 'NashTech', logo: '/static/images/homepage/partner-19.png' },
  { name: 'Moogento', logo: '/static/images/homepage/partner-20.png' },
  { name: 'SourceOne', logo: '/static/images/homepage/partner-21.png' },
];

export default { partners };
