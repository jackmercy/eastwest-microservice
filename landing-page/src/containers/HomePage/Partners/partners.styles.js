import { createUseStyles } from 'react-jss';

import { rem } from '../../../utils/jss';
import BREAKPOINTS from '../../../constants/BREAKPOINTS';
import COLORS from '../../../constants/COLORS';

export default createUseStyles({
  root: {
    backgroundColor: COLORS.bgGray01,
    maxWidth: '100%',
    paddingTop: rem(80),
    paddingBottom: rem(80),
  },
  slider: { maxWidth: rem(1350), margin: [0, 'auto'] },
  title: {
    fontWeight: 900,
    fontSize: rem(56),
    lineHeight: rem(72),
    marginTop: 0,
    marginBottom: rem(80),
    textAlign: 'center',
    transition: 'font-size .5s ease, margin .5s ease',
  },
  logo: { display: 'block', maxWidth: rem(160), maxHeight: rem(66) },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    root: { paddingTop: rem(48), paddingBottom: rem(48) },
    title: { fontSize: rem(32), lineHeight: rem(40), marginBottom: rem(48) },
  },
});
