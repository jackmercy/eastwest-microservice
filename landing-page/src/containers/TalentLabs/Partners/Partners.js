import { memo } from 'react';
import { useIntl } from 'react-intl';

import { Content } from '../../components';
import { partners } from '../../HomePage/Partners/partners.configs';

import useStyles from './partners.styles';

const Partners = () => {
  const intl = useIntl();
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Content className={classes.content}>
        <h2 className={classes.title} data-test="partners-title">
          {intl.formatMessage({ id: 'HOMEPAGE.PARTNERS.TITLE' })}
        </h2>
        <div className={classes.partners}>
          {partners.map((item) => (
            <div key={item.name} className={classes.partnerItem}>
              <img
                loading="lazy"
                alt={item.name}
                title={item.name}
                src={item.logo}
                className={classes.partnerlogo}
              />
            </div>
          ))}
        </div>
      </Content>
    </div>
  );
};

export default memo(Partners);
