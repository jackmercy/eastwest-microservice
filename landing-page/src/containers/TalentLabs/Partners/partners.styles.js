import { createUseStyles } from 'react-jss';

import { rem } from '../../../utils/jss';
import BREAKPOINTS from '../../../constants/BREAKPOINTS';
import COLORS from '../../../constants/COLORS';

export default createUseStyles({
  root: { backgroundColor: COLORS.bgGray01 },
  content: { paddingTop: rem(80), paddingBottom: rem(80) },
  title: {
    fontWeight: 900,
    fontSize: rem(56),
    lineHeight: rem(72),
    marginTop: 0,
    marginBottom: rem(80),
    textAlign: 'center',
    transition: 'font-size .5s ease, margin .5s ease',
  },
  partners: { display: 'flex', flexWrap: 'wrap' },
  partnerItem: {
    flex: '20% 0',
    maxWidth: '20%',
    marginBottom: rem(64),
    padding: [0, rem(8)],
    boxSizing: 'border-box',
    display: 'flex',
    alignItems: 'center',
  },
  partnerlogo: {
    flex: 'none',
    display: 'block',
    maxWidth: rem(160),
    maxHeight: rem(66),
    margin: [0, 'auto'],
  },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxLg}px)`]: {
    partnerItem: { flex: '25% 0', maxWidth: '25%', marginBottom: rem(48) },
  },
  [`@media only screen and (max-width: ${BREAKPOINTS.maxMd}px)`]: {
    content: { paddingTop: rem(32), paddingBottom: rem(32) },
    partnerItem: {
      flex: 'calc(100% / 3) 0',
      maxWidth: 'calc(100% / 3)',
      marginBottom: rem(32),
    },
  },
  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    title: { fontSize: rem(32), lineHeight: rem(40), marginBottom: rem(48) },
    partnerItem: { flex: 'auto' },
    partnerlogo: { maxWidth: rem(90) },
  },
});
