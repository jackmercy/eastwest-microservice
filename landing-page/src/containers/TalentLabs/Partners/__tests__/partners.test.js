import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Partners from '../Partners';

describe('Testing TalentLabs > Partners', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = mount(<Partners />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<Partners />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a title', () => {
    expect(wrapper.find('[data-test="partners-title"]')).toHaveLength(1);
  });
});
