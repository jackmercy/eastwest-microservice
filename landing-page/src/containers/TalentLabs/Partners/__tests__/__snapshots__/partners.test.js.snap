// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TalentLabs > Partners should render match snapshot 1`] = `
<div
  className="root-0-2-1"
>
  <Memo(Content)
    className="content-0-2-2"
  >
    <h2
      className="title-0-2-3"
      data-test="partners-title"
    >
      We are trusted by
    </h2>
    <div
      className="partners-0-2-4"
    >
      <div
        className="partnerItem-0-2-5"
        key="Corekees"
      >
        <img
          alt="Corekees"
          className="partnerlogo-0-2-6"
          loading="lazy"
          src="/static/images/homepage/partner-01.png"
          title="Corekees"
        />
      </div>
      <div
        className="partnerItem-0-2-5"
        key="RNW"
      >
        <img
          alt="RNW"
          className="partnerlogo-0-2-6"
          loading="lazy"
          src="/static/images/homepage/partner-02.png"
          title="RNW"
        />
      </div>
      <div
        className="partnerItem-0-2-5"
        key="KoiQuestion"
      >
        <img
          alt="KoiQuestion"
          className="partnerlogo-0-2-6"
          loading="lazy"
          src="/static/images/homepage/partner-03.png"
          title="KoiQuestion"
        />
      </div>
      <div
        className="partnerItem-0-2-5"
        key="Phoenix"
      >
        <img
          alt="Phoenix"
          className="partnerlogo-0-2-6"
          loading="lazy"
          src="/static/images/homepage/partner-04.png"
          title="Phoenix"
        />
      </div>
      <div
        className="partnerItem-0-2-5"
        key="Voicepick"
      >
        <img
          alt="Voicepick"
          className="partnerlogo-0-2-6"
          loading="lazy"
          src="/static/images/homepage/partner-05.png"
          title="Voicepick"
        />
      </div>
      <div
        className="partnerItem-0-2-5"
        key="TymeBank"
      >
        <img
          alt="TymeBank"
          className="partnerlogo-0-2-6"
          loading="lazy"
          src="/static/images/homepage/partner-06.png"
          title="TymeBank"
        />
      </div>
      <div
        className="partnerItem-0-2-5"
        key="Quadriz"
      >
        <img
          alt="Quadriz"
          className="partnerlogo-0-2-6"
          loading="lazy"
          src="/static/images/homepage/partner-07.png"
          title="Quadriz"
        />
      </div>
      <div
        className="partnerItem-0-2-5"
        key="Toiya"
      >
        <img
          alt="Toiya"
          className="partnerlogo-0-2-6"
          loading="lazy"
          src="/static/images/homepage/partner-08.png"
          title="Toiya"
        />
      </div>
      <div
        className="partnerItem-0-2-5"
        key="Dias Advisors"
      >
        <img
          alt="Dias Advisors"
          className="partnerlogo-0-2-6"
          loading="lazy"
          src="/static/images/homepage/partner-09.png"
          title="Dias Advisors"
        />
      </div>
      <div
        className="partnerItem-0-2-5"
        key="Duality"
      >
        <img
          alt="Duality"
          className="partnerlogo-0-2-6"
          loading="lazy"
          src="/static/images/homepage/partner-10.png"
          title="Duality"
        />
      </div>
      <div
        className="partnerItem-0-2-5"
        key="5QLive"
      >
        <img
          alt="5QLive"
          className="partnerlogo-0-2-6"
          loading="lazy"
          src="/static/images/homepage/partner-11.png"
          title="5QLive"
        />
      </div>
      <div
        className="partnerItem-0-2-5"
        key="Media 21"
      >
        <img
          alt="Media 21"
          className="partnerlogo-0-2-6"
          loading="lazy"
          src="/static/images/homepage/partner-12.png"
          title="Media 21"
        />
      </div>
      <div
        className="partnerItem-0-2-5"
        key="GambleCoin"
      >
        <img
          alt="GambleCoin"
          className="partnerlogo-0-2-6"
          loading="lazy"
          src="/static/images/homepage/partner-13.png"
          title="GambleCoin"
        />
      </div>
      <div
        className="partnerItem-0-2-5"
        key="Microad"
      >
        <img
          alt="Microad"
          className="partnerlogo-0-2-6"
          loading="lazy"
          src="/static/images/homepage/partner-14.png"
          title="Microad"
        />
      </div>
      <div
        className="partnerItem-0-2-5"
        key="Grad Greenhouse"
      >
        <img
          alt="Grad Greenhouse"
          className="partnerlogo-0-2-6"
          loading="lazy"
          src="/static/images/homepage/partner-15.png"
          title="Grad Greenhouse"
        />
      </div>
      <div
        className="partnerItem-0-2-5"
        key="Pycogroup"
      >
        <img
          alt="Pycogroup"
          className="partnerlogo-0-2-6"
          loading="lazy"
          src="/static/images/homepage/partner-16.png"
          title="Pycogroup"
        />
      </div>
      <div
        className="partnerItem-0-2-5"
        key="Seleno"
      >
        <img
          alt="Seleno"
          className="partnerlogo-0-2-6"
          loading="lazy"
          src="/static/images/homepage/partner-17.png"
          title="Seleno"
        />
      </div>
      <div
        className="partnerItem-0-2-5"
        key="Dagiac"
      >
        <img
          alt="Dagiac"
          className="partnerlogo-0-2-6"
          loading="lazy"
          src="/static/images/homepage/partner-18.png"
          title="Dagiac"
        />
      </div>
      <div
        className="partnerItem-0-2-5"
        key="NashTech"
      >
        <img
          alt="NashTech"
          className="partnerlogo-0-2-6"
          loading="lazy"
          src="/static/images/homepage/partner-19.png"
          title="NashTech"
        />
      </div>
      <div
        className="partnerItem-0-2-5"
        key="Moogento"
      >
        <img
          alt="Moogento"
          className="partnerlogo-0-2-6"
          loading="lazy"
          src="/static/images/homepage/partner-20.png"
          title="Moogento"
        />
      </div>
      <div
        className="partnerItem-0-2-5"
        key="SourceOne"
      >
        <img
          alt="SourceOne"
          className="partnerlogo-0-2-6"
          loading="lazy"
          src="/static/images/homepage/partner-21.png"
          title="SourceOne"
        />
      </div>
    </div>
  </Memo(Content)>
</div>
`;
